Automattic/jetpack
# Improve PricingTable component items

Description #
In an attempt to improve upon the fantastic work put in to [create](https://github.com/Automattic/jetpack/pull/25377) and [update](https://github.com/Automattic/jetpack/pull/26253) the new [PricingTable](https://automattic.github.io/jetpack-storybook/?path=/story/js-packages-components-pricing-table--default) component that will be used in the unification of standalone Jetpack product pricing pages, I have identified a number of minor updates that may present difficulties in particular cases and one potential area of improvement.

#### Updates:

- [x] The new discount label overlaps when `price` / `offPrice` values are too significant and/or when a currency symbol has been added
<img width="300" alt="Screen Shot 2022-09-21 at 15 10 36" src="https://user-images.githubusercontent.com/43220201/191837072-85d4ff9b-cc1d-4e6d-a25e-43561526da1a.png">

- [x] CSS for the main `Gridicon` component appears to be overriding the new `.popover-icon` element styles for the `PricingTable` component making the icon bold like the preceding text rather than `fill: =var(--jp-gray);` as intended. 
<img width="300" alt="Screen Shot 2022-09-21 at 15 16 39" src="https://user-images.githubusercontent.com/43220201/191837158-1330dc17-f18d-4e44-919e-67c73e548a3b.png">

- [x] ~~There is some odd/somewhat random behavior of popover tooltips sticking when hovering over them~~ 
   - The primary issue is being addressed in #26457 
- [x] Popovers can become hidden behind the primary admin sidebar menu.
<img width="300" alt="Screen Shot 2022-09-21 at 15 17 45" src="https://user-images.githubusercontent.com/43220201/191837233-19e78192-a354-497e-af82-f56719b53c3d.png">

- [x] ~~The TOS element does not remain centered on the columns and becomes cut off as we change screen sizes before entering the mobile view.~~
   - The primary issue is being addressed in #26509
<img width="600" alt="Screen Shot 2022-09-22 at 12 49 08" src="https://user-images.githubusercontent.com/43220201/191837941-6bea8e41-702c-40c6-8448-a5bb66d69301.png">

#### Improvements:
- [x] We can take advantage of the logic used in the `JetpackProductOffer` component in Calypso that introduces a discount label and automatically generates the content of the label based on a number of pricing-related variables. This will allow us to bring consistency between the various dashboards that may display a discount badge so there are no discrepancies in the calculations that may lead to unnecessary consumer confusion.
![Calypso Discount Label](https://user-images.githubusercontent.com/43220201/191839349-653f662c-158a-45b3-b403-151bb3bf3b22.jpeg)

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1202920335836514
- p1HpG7-hMo-p2

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






* Run `npm run storybook:dev` from the `js-packages/storybook` folder
* Change the default `price`, `offPrice`, and `currency` values and verify the promo label reacts accordingly
* Ensure the TOS component remains centered to columns 2 and 3 while decreasing viewport width until the mobile view is activated
* Play with the various options and ensure the components handles the changes as expected
* Load this PR using the Jetpack Beta plugin on a Jurassic Ninja site and verify that the icon tooltip popover handles and displays as expected
* Check the Figma file to ensure the design matches
